src/transformers/utils/generic.py

```diff
@@ -762,7 +762,7 @@ def torch_int(x):
 
     import torch
 
-    return x.to(torch.int64) if torch.jit.is_tracing() else int(x)
+    return x.to(torch.int64) if torch.jit.is_tracing() and isinstance(x, torch.Tensor) else int(x)
```

Comment:
Here you can use the [torch_int](https://github.com/huggingface/transformers/blob/cd0935dd55dcc1ba8d20305378d5f4e11126d225/src/transformers/utils/generic.py#L756) utility instead

Reply:
do you mean ensuring `shift_size` is a `torch.Tensor` when jit tracing in the modeling directly ?  https://github.com/huggingface/transformers/blob/7f91f168a14c7a305252675730b3248a1c987944/src/transformers/models/swin/modeling_swin.py#L643